x-pack/plugins/apm/public/components/app/service_details/service_icons/index.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLoadingSpinner } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { useUrlParams } from '../../../../context/url_params_context/use_url_params';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { getAgentIcon } from '../../../shared/AgentIcon/get_agent_icon';
+import { CloudDetails } from './cloud_details';
+import { ContainerDetails } from './container_details';
+import { IconPopover } from './icon_popover';
+import { ServiceDetails } from './service_details';
+import { ServiceDetailsContextProvider } from './service_details_context';
+
+interface Props {
+  serviceName: string;
+}
+
+const cloudIcons: Record<string, string> = {
+  gcp: 'logoGCP',
+  aws: 'logoAWS',
+  azure: 'logoAzure',
+};
+
+function getCloudIcon(provider?: string) {
+  if (provider) {
+    return cloudIcons[provider];
+  }
+}
+
+export function ServiceIcons({ serviceName }: Props) {
+  const {
+    urlParams: { start, end },
+    uiFilters,
+  } = useUrlParams();
+
+  const { data: icons, status: iconsFetchStatus } = useFetcher(
+    (callApmApi) => {
+      if (serviceName && start && end) {
+        return callApmApi({
+          endpoint: 'GET /api/apm/services/{serviceName}/icons',
+          params: {
+            path: { serviceName },
+            query: { start, end, uiFilters: JSON.stringify(uiFilters) },
+          },
+        });
+      }
+    },
+    [serviceName, start, end, uiFilters]
+  );
+
+  const isLoading =
+    !icons &&
+    (iconsFetchStatus === FETCH_STATUS.LOADING ||
+      iconsFetchStatus === FETCH_STATUS.PENDING);
+
+  const cloudIcon = getCloudIcon(icons?.cloud);
+  let containerIcon;
+  if (icons?.container) {
+    containerIcon =
+      icons.container === 'Kubernetes' ? 'logoKubernetes' : 'logoDocker';
+  }
+
+  return (
+    <ServiceDetailsContextProvider serviceName={serviceName}>
```

Comment:
Context is great when the same data is consumed by different components it's not feasible to lift the data fetching to a shared parent component that can pass it down.

However in this case I don't see why we can't do that. Afaict the consumers are immediate children of this component so will be simple to fetch the data here and pass it down.